lib/ui/viewer/gallery/gallery_app_bar_widget.dart

```diff
@@ -225,4 +256,7 @@ class _GalleryAppBarWidgetState extends State<GalleryAppBarWidget> {
       showGenericErrorDialog(context);
     }
   }
+
+  bool _isNotHiddenAndIsShared(Collection c) =>
+      (c.sharees.isNotEmpty || c.publicURLs.isNotEmpty) && !c.isArchived();
```

Comment:
Similar to `isArchive`, we can define `isShared` method in the collection and use that. No need to this additional `_isNotHiddenAndIsShared` method.